src/lightning/data/streaming/dataset.py

```diff
@@ -137,9 +125,11 @@ def _create_cache(self, worker_env: _WorkerEnv) -> Cache:
 
     def _create_shuffler(self, cache: Cache) -> Shuffle:
         seed = self.seed
+        drop_last = self.drop_last
         if self._state_dict is not None:
-            seed = self._state_dict[str(cache.rank)]["seed"]
-        return FullShuffle(cache, seed, self.drop_last) if self.shuffle else NoShuffle(cache, seed, self.drop_last)
+            seed = self._state_dict[-1]["seed"]
```

Comment:
Can we guarantee that the `self._state_dict` here always contains the key "seed", "drop_last", etc? And why is it correct to choose the one from the last process (-1)?

Reply:
`-1` meant this was the information from the last restart. Right now, we don't allow yet for you to change the metadata. In a future PR, I will add support for you to change the world_size, batch_size, num_workers.